app/routes/endemics/date-of-visit-ms.js

```diff
@@ -45,254 +40,138 @@ const previousPageUrl = (latestVetVisitApplication, typeOfReview, previousClaims
-      mainMessage.text =
-        'There must be no more than 10 months between your reviews and follow-ups.'
-      mainMessage.url =
-        'https://www.gov.uk/guidance/farmers-how-to-apply-for-funding-to-improve-animal-health-and-welfare#timing-of-reviews-and-follow-ups'
-      backToPageMessage =
-        'Enter the date the vet last visited your farm for this follow-up.'
-      break
-    case dateOfVetVisitExceptions.endemicsWithin10:
-      mainMessage.text =
-        'There must be at least 10 months between your follow-ups.'
-      mainMessage.url =
-        'https://www.gov.uk/guidance/farmers-how-to-apply-for-funding-to-improve-animal-health-and-welfare#timing-of-reviews-and-follow-ups'
-      backToPageMessage =
-        'Enter the date the vet last visited your farm for this follow-up.'
-      break
-    case dateOfVetVisitExceptions.claimEndemicsBeforeReviewPayment:
-      mainMessage.text =
-        'Your review claim must have been approved before you claim for the follow-up that happened after it.'
-      mainMessage.url =
-        'https://www.gov.uk/guidance/farmers-how-to-apply-for-funding-to-improve-animal-health-and-welfare#timing-of-reviews-and-follow-ups'
-      backToPageMessage =
-        'Enter the date the vet last visited your farm for this follow-up.'
-      break
+  const now = new Date()
+
+  if (dateOfVisit > now) {
+    return {
+      errorSummary: [{
+        text: `Error: The date of ${reviewOrFollowUpText} must be in the past`,
+        href: '#visit-date-day'
+      }],
+      inputsInError: { day: true, month: true, year: true }
+    }
+  }
+
+  if (new Date(request.payload.dateOfAgreementAccepted) > dateOfVisit) {
+    return {
+      errorSummary: [{
+        text: `Error: The date of ${reviewOrFollowUpText} cannot be before the date your agreement began`,
+        href: '#visit-date-day'
+      }],
+      inputsInError: { day: true, month: true, year: true }
+    }
   }
 
-  return { mainMessage, backToPageMessage }
+  return {
+    errorSummary: [],
+    inputsInError: { day: false, month: false, year: false }
+  }
 }
 
 const getHandler = {
   method: 'GET',
   path: pageUrl,
   options: {
     handler: async (request, h) => {
-      const { dateOfVisit, latestEndemicsApplication, typeOfReview, latestVetVisitApplication, previousClaims, typeOfLivestock } =
+      const { dateOfVisit, latestVetVisitApplication: oldWorldApplication, typeOfReview, latestVetVisitApplication, previousClaims, typeOfLivestock } =
         session.getEndemicsClaim(request)
       const { isReview } = getReviewType(typeOfReview)
       const reviewOrFollowUpText = isReview ? 'review' : 'follow-up'
 
-      return h.view(endemicsDateOfVisit, {
-        dateOfAgreementAccepted: new Date(latestEndemicsApplication.createdAt)
-          .toISOString()
-          .slice(0, 10),
+      return h.view(`${endemicsDateOfVisit}-ms`, {
+        dateOfAgreementAccepted: new Date(oldWorldApplication.createdAt).toISOString().slice(0, 10),
```

Comment:
Realise it was here already but we need to be careful with `.toISOString()` as it returns UTC rather than local date. So we open ourselves to odd edge cases such as claims submitted after 11pm in summertime showing as the wrong day.

Better to work with localeStrings. Something like this should ensure you always get the true date.
```
date.toLocaleString('en-GB', { year: 'numeric', month: 'numeric', day: 'numeric'})
  .split('/')
  .reverse()
  .join('-')
```
We have a few different approaches to this problem lying around, maybe we should talk about consolidating a set of common functions. Not for this PR though.

Reply:
Good call yeah - I copied across your suggestion and it seemed to be working fine. Then I had a closer look at what we are actually using this for, and I think it can be removed. The GET handler for the page seems to get the old world application date created, and then stores it in the page as a hidden input, and it only gets used on the POST handler when an error happens.

But we dont need to actually do this, because when we're in the handler we have the old world application stuff in the session - we dont need to get it from the form submission. I might be missing something here but it seems entirely needless to me.